client/src/pages/SeriesDetails.tsx

```diff
@@ -0,0 +1,139 @@
+import { useEffect, useState } from 'react';
+import { Smiskis } from '../../../server/lib/data';
+import { useNavigate, useParams } from 'react-router-dom';
+
+export function SeriesDetails() {
+  const [seriesDetails, setSeriesDetails] = useState<Smiskis[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [quantity, setQuantity] = useState(1);
+  const [err, setErr] = useState<unknown>();
+  const { seriesId } = useParams();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    async function getSeriesDetails() {
+      try {
+        const response = await fetch(`/api/series/${seriesId}/smiskis`);
+        if (!response.ok) throw new Error(`Response status ${response.status}`);
+        const smiskis = (await response.json()) as Smiskis[];
+        setSeriesDetails(smiskis);
+      } catch (err) {
+        setErr(err);
+      } finally {
+        setLoading(false);
+      }
+    }
+    getSeriesDetails();
+  }, [seriesId]);
+
+  function handleAddQuantity() {
+    const addQuantity = quantity + 1;
+    setQuantity(addQuantity);
+  }
+
+  function handleSubtractQuantity() {
+    let subtractQuantity = quantity - 1;
+    if (subtractQuantity < 1) {
+      subtractQuantity = 1;
+    }
+    setQuantity(subtractQuantity);
+  }
+
+  async function handleAddToCart() {
+    try {
+      const response = await fetch('/api/shoppingCartItems', {
+        method: 'POST',
+      });
+      if (!response.ok) throw new Error(`Response status ${response.status}`);
+      alert('Added to cart!');
+    } catch (err) {
+      setErr(err);
+    }
+  }
+
+  const seriesDetailsBanner = (seriesId: string | undefined) => {
+    switch (seriesId) {
+      case '1':
+        return '/images/series1cover2.webp';
+      case '2':
+        return '/images/series2cover2.png';
+      case '3':
+        return '/images/series3cover2.webp';
+      case '4':
+        return '/images/series4cover22.webp';
+    }
+  };
```

Comment:
Aren't these image paths stored in your database? I would access the ones from there, rather than hard coding these